### PR TITLE
Remove pointerdb.Server

### PR DESCRIPTION
**Pull request by egonelbre**
The code has migrated to either using `pointerdb.Service` and `metainfo`, so  pointerdb server and allocation signer aren't necessary anymore.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/pointerdb/pointerdb_test.go [hunk 1]
```diff
@@ -1,472 +0,0 @@
-// Copyright (C) 2019 Storj Labs, Inc.
-// See LICENSE for copying information.
-
-package pointerdb_test
```
Comment (egonelbre) [inline pkg/pointerdb/pointerdb_test.go]: Not sure what to do about these tests. We already have them covered elsewhere, however with different approaches.

Comment (egonelbre) [reply pkg/pointerdb/pointerdb_test.go]: @kaloyan-raev, @mniewrzal ?

Comment (mniewrzal) [reply pkg/pointerdb/pointerdb_test.go]: Get and Delete tests can be deleted for me. I'm looking into List and Put tests.

Comment (mniewrzal) [reply pkg/pointerdb/pointerdb_test.go]: Regarding Put tests I would add at least one test for `Number of valid pieces is lower then success threshold` somewhere. Testplanet uplink.Upload with custom redundancy scheme. List tests would be nice to move/adjust I guess. We can upload test data as inline pointers.

Comment (egonelbre) [reply pkg/pointerdb/pointerdb_test.go]: For List we do have https://github.com/storj/storj/blob/master/pkg/metainfo/kvmetainfo/objects_test.go#L293 as tests. However that is at the db level.

Comment (mniewrzal) [reply pkg/pointerdb/pointerdb_test.go]: Ok, maybe its enough. Additional value with pointerdb_test is that it's trying to check some special characters (at least at basic level).

Comment (egonelbre) [reply pkg/pointerdb/pointerdb_test.go]: It looks like we are not verifying the "lower than success threshold" at the moment.

Comment (egonelbre) [reply pkg/pointerdb/pointerdb_test.go]: I created a separate issue for the "Number of valid pieces is lower then success threshold".

Comment (egonelbre) [reply pkg/pointerdb/pointerdb_test.go]: And implemented basic utf-8 list tests.

**Pull request merged**